File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/datasource/NetworkDataSource.java
package com.didichuxing.doraemonkit.kit.performance.datasource;

import com.didichuxing.doraemonkit.kit.network.NetworkManager;
import com.didichuxing.doraemonkit.kit.network.utils.ByteUtil;
import com.didichuxing.doraemonkit.kit.performance.widget.LineData;

/**
 * @desc: 抓包数据源
 */
public class NetworkDataSource implements IDataSource {
    private static final String TAG = "NetworkDataSource";
    private long latestTotalLength = -1;

    @Override
    public LineData createData() {
        long diff = 0;
        long totalSize = NetworkManager.get().getTotalSize();
        if (latestTotalLength >= 0) {
            diff = totalSize - latestTotalLength;
            if (diff < 0) {
                diff = 0;
            }
        }
        latestTotalLength = totalSize;
        if (diff == 0) {
            return LineData.obtain((float) Math.ceil(diff / 1024f), null);
        } else {
            return LineData.obtain((float) Math.ceil(diff / 1024f), ByteUtil.getPrintSize(diff));
        }
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/manager/datasource/NetworkDataSource.kt
package com.didichuxing.doraemonkit.kit.performance.manager.datasource

import com.didichuxing.doraemonkit.kit.network.NetworkManager.Companion.get
import com.didichuxing.doraemonkit.kit.network.utils.ByteUtil
import com.didichuxing.doraemonkit.kit.performance.manager.datasource.IDataSource
import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData
import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData.Companion.obtain
/**
 * @desc: 抓包数据源
 */
class NetworkDataSource : IDataSource {
    private var latestTotalLength: Long = -1
    override fun createData(): LineData {
        var diff: Long = 0
        val totalSize = get().totalSize
        if (latestTotalLength >= 0) {
            diff = totalSize - latestTotalLength
            if (diff < 0) {
                diff = 0
            }
        }
        latestTotalLength = totalSize
        return if (diff == 0L) {
            obtain(Math.ceil(diff / 1024f.toDouble()).toFloat(), null)
        } else {
            obtain(Math.ceil(diff / 1024f.toDouble()).toFloat(), ByteUtil.getPrintSize(diff))
        }
    }

    companion object {
        private const val TAG = "NetworkDataSource"
    }
}